pkg/api/object.go

```diff
@@ -89,13 +83,155 @@ func (h *Handler) GetObjectStoreConnectionInfo(w http.ResponseWriter, r *http.Re
 	s3Info := model.ObjectStoreS3Info{
 		Host:       host,
 		IPEndpoint: ipEndpoint,
-		AccessKey:  accessKey,
-		SecretKey:  secretKey,
 	}
 
 	FormatJsonResponse(w, s3Info)
 }
 
+// ListUsers lists the users of the object store in this cluster.
+// GET
+// /objectstore/users
+func (h *Handler) ListUsers(w http.ResponseWriter, r *http.Request) {
+	userNames, err := rgw.ListUsers(h.context, h.AdminCommandRunner)
+	if err != nil {
+		logger.Errorf("Error listing users: %+v", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	users := []model.ObjectUser{}
+	for _, userName := range userNames {
+		user, err := rgw.GetUser(h.context, h.AdminCommandRunner, userName)
+		if err != nil {
+			logger.Errorf("Error listing users: %+v", err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		users = append(users, user)
+	}
+
+	FormatJsonResponse(w, users)
+}
+
+// GetUser gets the passed users info from the object store in this cluster.
+// GET
+// /objectstore/users/{USER_ID}
+func (h *Handler) GetUser(w http.ResponseWriter, r *http.Request) {
+	id := mux.Vars(r)["id"]
+
+	user, err := rgw.GetUser(h.context, h.AdminCommandRunner, id)
+	if err != nil {
+		logger.Errorf("Error gettin user (%s): %+v", id, err)
+
+		if err.Error() == "user not found" {
+			w.WriteHeader(http.StatusNotFound)
+		} else {
+			w.WriteHeader(http.StatusInternalServerError)
+		}
+		return
+	}
+
+	FormatJsonResponse(w, user)
+}
+
+// CreateUser will create a new user from the passed info in the object store in this cluster.
+// POST
+// /objectstore/users
+func (h *Handler) CreateUser(w http.ResponseWriter, r *http.Request) {
+	var user model.ObjectUser
+
+	err := json.NewDecoder(r.Body).Decode(&user)
+	if err != nil {
+		logger.Errorf("Error parsing user: %+v", err)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	user, err = rgw.CreateUser(h.context, h.AdminCommandRunner, user)
+	if err != nil {
+		logger.Errorf("Error creating user: %+v", err)
+
+		if err.Error() == "user already exists" || err.Error() == "email already in use" || err.Error() == "displayName is required" || err.Error() == "userId cannot be empty" {
```

Comment:
again, checking for error messages is brittle. what about just returning a 400 bad request?